addons/sale/migrations/12.0.1.1/pre-migration.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2019 Eficent <http://www.eficent.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+_column_renames = {
+    'payment_transaction': [
+        ('sale_order_id', None),
+    ],
+    'sale_order': [
+        ('payment_tx_id', None),
+    ],
+}
+
+_field_renames = [
+    ('sale.order', 'sale_order', 'requested_date', 'commitment_date'),
+]
+
+
+def fill_sale_order_line_sections(cr):
+    """It's done here instead of post-migration to avoid
+    possible new rows added in the migration"""
+    cr.execute(
+        "ALTER TABLE sale_order_line ADD COLUMN display_type varchar",
+    )
+    openupgrade.logged_query(
+        cr, """
+        INSERT INTO sale_order_line (order_id, layout_category_id,
+            sequence, name, price_unit, product_uom_qty, customer_lead,
+            display_type, create_uid, create_date, write_uid, write_date)
+        SELECT sol.invoice_id, sol.layout_category_id,
```

Comment:
there is no invoice_id column in the sale_order_line table. You probably meant ```sol.order_id```, right?